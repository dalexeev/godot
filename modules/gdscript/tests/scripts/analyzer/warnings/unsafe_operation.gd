func test_no_exec():
	var hard_int: int = 1
	var variant_int: Variant = 1
	var weak_int = 1

	print(hard_int + hard_int)
	print(hard_int + variant_int)
	print(hard_int + weak_int)
	print(variant_int + variant_int)
	print(variant_int + weak_int)
	print(weak_int + weak_int)

	print(-hard_int)
	print(-variant_int)
	print(-weak_int)


func test():
	pass
