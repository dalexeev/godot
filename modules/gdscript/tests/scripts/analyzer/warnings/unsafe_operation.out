GDTEST_OK
>> WARNING
>> Line: 8
>> UNSAFE_OPERATION
>> The type of at least one operand is unknown in the operation "+".
>> WARNING
>> Line: 9
>> UNSAFE_OPERATION
>> The type of at least one operand is unknown in the operation "+".
>> WARNING
>> Line: 10
>> UNSAFE_OPERATION
>> The type of at least one operand is unknown in the operation "+".
>> WARNING
>> Line: 11
>> UNSAFE_OPERATION
>> The type of at least one operand is unknown in the operation "+".
>> WARNING
>> Line: 12
>> UNSAFE_OPERATION
>> The type of at least one operand is unknown in the operation "+".
>> WARNING
>> Line: 15
>> UNSAFE_OPERATION
>> The type of at least one operand is unknown in the operation "unary-".
>> WARNING
>> Line: 16
>> UNSAFE_OPERATION
>> The type of at least one operand is unknown in the operation "unary-".
>> WARNING
>> Line: 19
>> UNSAFE_OPERATION
>> The type of at least one operand is unknown in the operation "+".
>> WARNING
>> Line: 20
>> UNSAFE_OPERATION
>> The type of at least one operand is unknown in the operation "+".
